src/main/java/TaskList.java

```diff
@@ -0,0 +1,253 @@
+            while ((line = reader.readLine()) != null) {
+                processFileLine(line);
+            }
+            reader.close();
+        } catch (IOException | DukeException e) {
+            System.out.println("Error loading tasks from file: " + e.getMessage());
+        }
+    }
+
+    private void processFileLine(String line) throws DukeException {
+        // Parse the line and create tasks based on the format in the file
+        String[] parts = line.split(" \\| ");
+        if (parts.length != 3) {
+            throw new DukeException("Invalid task format in the file.");
+        }
+
+        String taskType = parts[0];
+        boolean isDone = parts[1].equals("1");
+        String description = parts[2];
+
+        switch (taskType) {
+            case "T":
+                // Check if the description contains date information
+                String[] todoParts = description.split(" \\(from: | to: ", 3);
+                if (todoParts.length == 3) {
+                    LocalDate fromDate = LocalDate.parse(todoParts[1]);
+                    LocalDate toDate = LocalDate.parse(todoParts[2].substring(0, todoParts[2].length() - 1));
+                    addTask(new TodoTask(todoParts[0], fromDate, toDate, isDone));
+                } else {
+                    throw new DukeException("Invalid todo task format in the file.");
+                }
+                break;
+            case "D":
+                // Check if the description contains date information
+                String[] deadlineParts = description.split(" \\(by: ", 2);
+                if (deadlineParts.length == 2) {
+                    LocalDate byDate = LocalDate.parse(deadlineParts[1].substring(0, deadlineParts[1].length() - 1));
+                    addTask(new DeadlineTask(deadlineParts[0], byDate, isDone));
+                } else {
+                    throw new DukeException("Invalid deadline task format in the file.");
+                }
+                break;
+            case "E":
+                // Check if the description contains date information
+                String[] eventParts = description.split(" \\(at: ", 2);
+                if (eventParts.length == 2) {
+                    LocalDate atDate = LocalDate.parse(eventParts[1].substring(0, eventParts[1].length() - 1));
+                    addTask(new EventTask(eventParts[0], atDate, isDone));
+                } else {
+                    throw new DukeException("Invalid event task format in the file.");
+                }
+                break;
+            default:
+                throw new DukeException("Unknown task type in file.");
+        }
+    }
+    public List<Task> getAllTasks() {
+        return tasks;
+    }
+
+    public int getTotalTasks() {
+        return tasks.size();
+    }
+
+    public Task getTask(int index) {
+        if (index >= 0 && index < tasks.size()) {
+            return tasks.get(index);
+        } else {
+            return null;
```

Comment:
Is there a reason why getTask() do not throw a DukeException specifying that invalid task index was inputted?